src/index.js

```diff
@@ -0,0 +1,168 @@
+import { differenceCiede2000 } from 'd3-color-difference';
+
+export default class toColor {
+  HUE_MAX = 360;
+
+  constructor(seed, options) {
+    this.options = options || {};
+    if (typeof seed === 'string' || typeof seed === 'number') {
+      this.seed = typeof seed === 'string' ? this._stringToInteger(seed) : seed;
+    } else {
+      throw new TypeError('Seed value must be a number or string');
+    }
+
+    this.known = [];
+  }
+
+  getColor(count = 0) {
+    const h = this._pickHue();
+    const s = this._pickSaturation(h);
+    const b = this._pickBrightness(h, s);
+    const hsl = this._HSVtoHSL(h, s, b);
+    const formatted = this._formatHSL(hsl);
+    const PASSABLE_DISTANCE = 50; 
+
+    // The larger `count` grows, we need to divide actual distance to avoid
+    // hitting a maxiumum call stack error.
+    const ACTUAL_DISTANCE = PASSABLE_DISTANCE / Math.pow(1.25, count);
+
+    // Detect color similarity. If values are too close to one another, call
+    // getColor until enough dissimilarity is acheived.
+    if (this.known.length &&
+        this.known.some(v => differenceCiede2000(v, formatted) < ACTUAL_DISTANCE)) {
+      count++
+      return this.getColor(count);
+    } else {
+      this.known.push(formatted);
+      // Apply modifiers after distribution check + regeneration to ensure 
+      // colors with brightness/saturation adjustments remain the same.
+      return this._colorWithModifiers(h, s, b) 
+    }
+  }
+
+  _colorWithModifiers = (h, s, b) => {
+    const clamp = (n, min, max) => n <= min ? min : n >= max ? max : n;
+    const { brightness, saturation } = this.options;
+
+    // Modify brightness/saturation if provided
+    s = saturation ? clamp(s + saturation, 0, 100) : s;
+    b = brightness ? clamp(b + brightness, 0, 100) : b;
+
+    // re-run conversion accounting for post modifications to s and b.
+    const hsl = this._HSVtoHSL(h, s, b);
+
+    return {
+      hsl: {
+        raw: hsl,
+        formatted: this._formatHSL(hsl)
+      }
+    };
+  }
+  
+  _formatHSL = hsl => `hsl(${hsl[0]}, ${hsl[1]}%, ${hsl[2]}%)`
+
+  _pickHue = () => {
+    let hue = this._pseudoRandom([0, this.HUE_MAX]);
+    const min = hue % this.HUE_MAX;
+    const max = (hue + 1) % this.HUE_MAX;
+
+    hue = this._pseudoRandom([min, max]);
+
+    // Red is on both ends of the color spectrum. Instead of storing red as two
+    // ranges, lookup is grouped in colorDictionary using negative numbers.
+    if (hue < 0) hue = this.HUE_MAX + hue;
+
+    return hue;
+  }
+
+  _pickSaturation = (h) => {
+    const saturationRange = this._getColorInfo(h)[2];
```

Comment:
You approach has drifted quite a bit from randomColor - do we still need to do saturation and brightness picking based on the color dictionary in the way randomColor does? Curious if we can simplify now that we're leaning more on the color difference algo to help us pick good colors.

Reply:
I think the premise of suitable ranges for saturation/brightness still holds up. Distribution is different from randomColor but I think staying within suitable range of color still makes sense and address areas where dark colors are ignored.